Source/Scene/Cesium3DTileset.js

```diff
@@ -70,19 +72,25 @@ define([
         }
         //>>includeEnd('debug');
 
-        var tilesJson;
+        var tilesetUrl;
+        var baseUrl;
+
         if (getExtensionFromUri(url) === 'json') {
-            tilesJson = url;
-            url = getBaseUri(url);
+            tilesetUrl = url;
+            baseUrl = getBaseUri(url);
+        } else if (isDataUri(url)) {
+            tilesetUrl = url;
+            baseUrl = '';
         } else {
-            url = appendForwardSlash(url);
-            tilesJson = url + 'tiles.json';
+            baseUrl = appendForwardSlash(url);
+            tilesetUrl = baseUrl + 'tiles.json';
         }
 
         this._url = url;
```

Comment:
I think the original version I coded was a mistake.  This should just be used for the `url` getter, which should return the same url passed to the constructor function.